packages/db/tests/models/test-granule-model.js

```diff
@@ -402,6 +402,41 @@ test('GranulePgModel.upsert() will allow a running status to replace a non-runni
   t.is(record.status, 'running');
 });
 
+test('GranulePgModel.upsert() will allow a queued status to replace a non-queued status for different execution', async (t) => {
```

Comment:
For completeness, I think we would also want the unit tests to prove that a "completed" status can replace a "queued" status and that a "running' status can replace a "queued" status